ivy/stateful/activations.py

```diff
@@ -275,24 +275,34 @@ def _forward(self, x):
 
 
 class Tanh(Module):
-    def __init__(self):
-        """Apply the TANH activation function."""
+    def __init__(self, complex_mode: Literal["split", "magnitude", "jax"] = "jax"):
+        """
+        Apply the TANH activation function.
+
+        Parameters
+        ----------
+        complex_mode
+            Specifies how to handle complex input.
+        """
+        self._complex_mode = complex_mode
         Module.__init__(self)
 
-    def _forward(self, x):
+    def _forward(self, x, complex_mode=None):
```

Comment:
let's remove the `complex_mode` parameter from `_forward` as we shouldn't need to pass a separate value for `complex_mode` for each forward pass. [This](https://github.com/unifyai/ivy/pull/21902) PR fixes it for the rest of the stateful classes

Reply:
Wasn't aware of that PR but that does make a lot more sense